src/table/pagination-table/components/TableWrapper.tsx

```diff
@@ -48,6 +52,10 @@ function TableWrapper(props: TableWrapperProps) {
     tableContainerRef.current?.scrollTo(0, 0);
   }, [pageInfo, totalRowCount]);
 
+  useLayoutEffect(() => {
+    setBorderHeight(paginationTableRef.current?.getBoundingClientRect?.()?.height);
```

Comment:
Does this `paginationTableRef.current?.getBoundingClientRect?.()?.height` ever return `undefined`? If so you, probably do 1 extra render

Reply:
Right, should do like this
```
  useLayoutEffect(() => {
    const tableHeight = paginationTableRef.current?.getBoundingClientRect?.()?.height;
    if (tableHeight) setBorderHeight(tableHeight);
  }, [rows.length, headHeight]);
```